domaci_bandit.py

```diff
@@ -98,54 +99,113 @@ def train(bandits_no=5, attempts_no=5000, alpha=0.1, epsilon=0.1, plotting=True)
         plt.scatter(range(len(q)), q, marker=".")
         plt.scatter(range(len(q)), [b.mean for b in env.bandits], marker="x")
 
-    return env.bandits, rewards, q
+    return env, rewards, q
 
 
 # Zadatak 1 : pokrenuti trening za razlicite vrednosti epsilon, prikazati rezultate i izvesti zakljucak o nagibu krive
 def prvi_zadatak():
     plt.figure(figsize=(14, 3))
     for i in range(3):
-        envi, rew, q1 = train(epsilon=10**(-i-1), plotting=False)
-        plot_e(3, i+1, envi, rew)
-        plt.title(10**(-i-1))
+        envi, rew, q1 = train(epsilon=10 ** (-i - 1), plotting=False)
+        plot_e(3, i + 1, envi, rew)
+        plt.title(10 ** (-i - 1))
 
     plt.show()
 
+
 def plot_e(n, i, envi, rew):
     plt.subplot(1, n, i)
     g = np.cumsum(rew)
     max_r = max([b.mean for b in envi])
     plt.plot(g, "r")
-    plt.plot(np.cumsum(max_r * np.ones(len(g))), "b")  
+    plt.plot(np.cumsum(max_r * np.ones(len(g))), "b")
+
 
 # Zakljucak prvog zadatka jeste da bismo najbolje prosli (dobili najvecu kumulativnu nagradu) da smo koristili greedy politiku sve vreme.
 # Prilikom smanjenja vrednosti epsilona, cesce je birana masina sa najvecom srednjom vrednosti (samim tim i veca nagarda) te je nagib
 # krive kumulativne nagrade rastao.
 
 
 # Zadatak 2 : Sa naucenim q i epsilon = 0, pustiti 100 iteracija
-def drugi_zadatak():
-    plt.figure(figsize=(8, 3))
-
-    envi, rew, q1 = train(plotting=False)
-    plot_e(2, 1, envi, rew)
-    plt.title("5000 iteracija")
-
-    plt.subplot(1, 2, 2)
-    test_len = 100
-    reward_q1 = [max(q1) for _ in range(test_len)]  # Uvek ce biti ista nagrada
+def drugi_zadatak(plotting=False):
+    env, rewards, q = train(plotting=False)
+    learned_q = q.copy()
+
+    # 100 iteracija sa epsilon = 0 (greedy metoda)
+    greedy_rewards = []
+    for _ in range(100):

```

Comment:
ceo ovaj for je jusles i zbog njega ceo prvi zadatak ne radi trn. 
choose_greedy)action(q) vraca indeks elementa sa najvecim reward, vratice isti indeks 100 puta.
env.take_action() ode u metodu pull_leaver Bandita, tako da bih ja stavila da se za taj indeks zove pull_leaver samo, posto to ne utice na env i samim tim se i prvi zadatak popravlja. (u tom slucaju promeniti return funckije tran u env.bandits)

Reply:
Pa nije jusles simulira realan rad koda, ne mozes samo dodji i pejstovati istu vrednost 100 puta u niz jer je "to to"

Reply:
A popravio sam ti onaj prvi zadatak onda da je env.bandits a ne env